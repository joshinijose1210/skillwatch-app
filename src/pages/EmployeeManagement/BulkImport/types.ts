export type stepSectionType = {
    employees: JSX.Element;
    kpis: JSX.Element;
};

export type bulkImportTypes = 'employees' | 'kpis';

export interface LocationState {
    bulkImportType: bulkImportTypes;
}
