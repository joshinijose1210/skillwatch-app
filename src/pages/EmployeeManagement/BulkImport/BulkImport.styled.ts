import styled from 'styled-components';

export const StyledBox = styled.div`
    border-radius: 1rem;
    margin: 1rem 0;
    width: 100%;
    display: flex;
    justify-content: space-between;
`;

export const BulkImportContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
    min-height: 80vh;
    height: fit-content;
`;
export const TimelineWrapper = styled.div`
    padding: 2rem;
    text-align: center;
    width: 50%;
    background-color: ${({ theme }) => theme.colors.grey[50]};
    border: 1px solid #c7d0d8;
`;
export const TimelineList = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 3rem;
    color: ${({ theme }) => theme.colors.grey[600]};
`;

export const ListWrapper = styled.div`
    display: flex;
    gap: 1rem;
    align-items: flex-start;
`;

export const SeparatorLine = styled.div`
    height: 100%;
`;
export const CSVWrapper = styled.div`
    width: 50%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    min-width: 480px;
    border: 1px solid ${({ theme }) => theme.colors.grey[300]};
    border-left: 0;
    min-height: 450px;
`;

export const DropZoneWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 3rem;
`;
export const DropZoneContainer = styled.div`
    border: 0.5rem ${({ theme }) => theme.colors.grey[300]};
    border-style: dashed;
    height: 20rem;
    width: 45rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
`;

export const DropFileInput = styled.input`
    display: none;
    cursor: pointer;
`;
export const DropFileText = styled.label`
    color: ${({ theme }) => theme.colors.blue[400]};
    cursor: pointer;
`;
export const ErrorText = styled.div`
    color: rgb(255, 0, 51);
    cursor: pointer;
`;
export const StyledImage = styled.img`
    width: 10rem;
    height: 10rem;
`;

export const CloseButton = styled.span`
    margin-left: 2rem;
    color: black;
    cursor: pointer;
`;

export const ItalicBold = styled.i`
    font-weight: bold;
`;

export const StepsList = styled.ul`
    list-style: none;
    color: ${({ theme }) => theme.colors.grey[600]};
`;

export const Step = styled.li<{ isSubItem?: boolean }>`
    margin-top: 1rem;
    position: relative;
    &::before {
        display: inline-block;
        content: '';
        border-radius: 0.375rem;
        height: 0.75rem;
        width: 0.75rem;
        margin-right: 0.5rem;
        position: absolute;
        left: -35px;
        top: 5px;
        background-color: ${({ theme, isSubItem }) => (!isSubItem ? theme.colors.grey[300] : 'none')};
        border: ${({ theme, isSubItem }) => (isSubItem ? `2px solid ${theme.colors.grey[300]}` : 'none')};
    }
    ::marker {
        top: 0;
    }
`;
