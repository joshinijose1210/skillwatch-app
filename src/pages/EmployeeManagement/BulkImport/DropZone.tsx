import { Button } from '@medly-components/core';
import upload from '../../../constants/images/logos/upload1.png';
import { CloseButton, DropFileInput, DropFileText, DropZoneContainer, DropZoneWrapper, ErrorText, StyledImage } from './BulkImport.styled';
import useBulkImport from './useBulkImport';

const DropZone = () => {
    const { fileName, handleBulkUpload, importError, reset, isLoading, getRootProps, getInputProps } = useBulkImport();

    return (
        <DropZoneWrapper>
            {fileName ? (
                <DropZoneContainer>
                    <StyledImage src={upload} alt="upload" />
                    <em>
                        <i>{fileName}</i>
                        <CloseButton data-testid="resetFile" onClick={reset}>
                            x
                        </CloseButton>
                    </em>
                </DropZoneContainer>
            ) : (
                <DropZoneContainer {...getRootProps()}>
                    <StyledImage src={upload} alt="upload" />
                    <DropFileText id="button"> Drop File or Upload File here.</DropFileText>
                    <DropFileInput data-testid="dropZone" {...getInputProps()} />
                </DropZoneContainer>
            )}
            {importError && <ErrorText>{importError}</ErrorText>}
            <Button data-testid="importBtn" onClick={handleBulkUpload} isLoading={isLoading} disabled={!fileName || !!importError}>
                Start Import
            </Button>
            <em>(Only .csv file will be accepted)</em>
        </DropZoneWrapper>
    );
};
export default DropZone;
