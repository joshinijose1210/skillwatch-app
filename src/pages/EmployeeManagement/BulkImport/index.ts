import BulkImport from './BulkImport';

export default BulkImport;
