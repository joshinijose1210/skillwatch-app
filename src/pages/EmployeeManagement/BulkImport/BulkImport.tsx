import { PageContent } from '@components';
import ListHeader from '@components/reusableComponents/ListHeader';
import { routeConstants } from '@constants';
import { Button, Text } from '@medly-components/core';
import {
    DoneAllIcon,
    DownloadIcon,
    GroupIcon,
    LibraryAddIcon,
    MailOutlineIcon,
    TouchAppIcon,
    UploadIcon,
    SaveIcon
} from '@medly-components/icons';
import { FlexStartDiv, StyledGoBackButton } from '@pages/RolesAndPermissions/RolesAndPermissions.styled';
import { FC } from 'react';
import {
    BulkImportContainer,
    CSVWrapper,
    ListWrapper,
    Step,
    StepsList,
    StyledBox,
    TimelineList,
    TimelineWrapper
} from './BulkImport.styled';
import DropZone from './DropZone';
import { stepSectionType } from './types';
import useBulkImport from './useBulkImport';

const BulkImport: FC = () => {
    const { handleDownloadFile, path, goBackHandle, bulkImportType } = useBulkImport();
    const stepsSection: stepSectionType = {
        employees: (
            <TimelineList>
                <Text textVariant="h4" textWeight="Medium">
                    Steps To Bulk Import Employees
                </Text>
                <ListWrapper>
                    <DoneAllIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Make sure at least 1 employee with a Manager role is manually added to the application.
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <DownloadIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Download the CSV template.
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <GroupIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Start adding employees to the CSV template:
                        <StepsList>
                            <Step>While adding the employees, make sure employees with Manager roles are added first. </Step>
                            <Step>Enter valid data for all the employees </Step>
                            <Step>You can set the status of the employee as Active by entering Y in the ‘Active’ column. </Step>
                            <Step>You can set the status of the employee as Inactive by entering N in the ‘Active’ column. </Step>
                            <Step>
                                You can set the role according to the roles listed in the application (example: HR, Manager, Employee), in
                                the ‘Role’ column depending on the different module/feature permissions you would like to give an employee.
                            </Step>
                            <Step>Maximum 500 employees can be added at a time. </Step>
                            <Step>Once you enter all the valid data, please save the CSV template. </Step>
                        </StepsList>
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <UploadIcon size="M" />

                    <Text textVariant="body1" textWeight="Regular">
                        Upload the CSV template.
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <TouchAppIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Click on the Start Import button.
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <MailOutlineIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Auto generated Invitation Email will be sent to employees.
                    </Text>
                </ListWrapper>
            </TimelineList>
        ),
        kpis: (
            <TimelineList>
                <Text textVariant="h4" textWeight="Medium">
                    Steps To Bulk Import KPIs
                </Text>
                <ListWrapper>
                    <DownloadIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Step 1: Download the CSV template.
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <LibraryAddIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Step 2: Start adding the following details in the CSV template.
                        <StepsList>
                            <Step>
                                Add KRA: Please make sure they already exist in the system and no grammatical/spelling mistakes are made.
                            </Step>
                            <Step>Add KPI Title: A maximum of 500 KPIs can be added at a time.</Step>
                            <Step>Add KPI Description: A maximum limit of 1000 characters.</Step>
                            <Step>Add Status: Enter Y to keep the KPI Active and enter N to keep the KPI Inactive.</Step>
                            <Step>
                                Add Department [Team 1 (Designation)]: Please make sure they already exist in the system and no
                                grammatical/spelling mistakes are made.
                            </Step>
                            <Step>
                                To separate two or more designations for a team, use a separator icon ( | ). Example:
                                <StepsList>
                                    <Step isSubItem>Product Management [Business Analyst (BA-1 | BA-2)]</Step>
                                    <Step isSubItem>
                                        Human Resource [People Experience (People Experience Associate | People Experience Manager)]
                                    </Step>
                                    <Step isSubItem>Engineering [Backend (SDE-1)]</Step>
                                </StepsList>
                            </Step>
                            <Step>
                                If you want to add the same KPIs for multiple teams you can repeat the Department [Team 2 (Designation)]
                                format in the next column and so on.
                            </Step>
                        </StepsList>
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <SaveIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Step 3: Once you enter all the valid data, save the CSV template file in CSV format.
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <UploadIcon size="M" />

                    <Text textVariant="body1" textWeight="Regular">
                        Step 4: Upload the CSV template.
                    </Text>
                </ListWrapper>
                <ListWrapper>
                    <TouchAppIcon size="M" />
                    <Text textVariant="body1" textWeight="Regular">
                        Step 5: Click on the Start Import button.
                    </Text>
                </ListWrapper>
            </TimelineList>
        )
    };
    return (
        <PageContent>
            {path.includes(routeConstants.firstEmployeeBulkImportRedirect) && (
                <FlexStartDiv>
                    <StyledGoBackButton variant="flat" onClick={goBackHandle}>
                        &lt; Go Back
                    </StyledGoBackButton>
                </FlexStartDiv>
            )}
            <ListHeader title="Bulk Import" />
            <StyledBox>
                <BulkImportContainer>
                    <TimelineWrapper>{stepsSection[bulkImportType]}</TimelineWrapper>
                    <CSVWrapper>
                        <ListWrapper>
                            <Button variant="outlined" onClick={handleDownloadFile} data-testid="templateDownload">
                                <DownloadIcon size="M" />
                                Download csv template
                            </Button>
                        </ListWrapper>
                        <DropZone />
                    </CSVWrapper>
                </BulkImportContainer>
            </StyledBox>
        </PageContent>
    );
};
export default BulkImport;
