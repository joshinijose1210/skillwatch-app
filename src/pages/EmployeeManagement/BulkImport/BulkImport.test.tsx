import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import BulkImport from '.';
import MockAdapter from 'axios-mock-adapter';
import * as reactRouterDom from 'react-router-dom';
import { addEmployeeMockData } from '@mocks/addEmployee';
import axios from 'axios';
import { apiInstance } from '@utils';
import { routeConstants } from '@constants';
const mock = new MockAdapter(axios);
const mockInsta = new MockAdapter(apiInstance);
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useNavigate: () => mockNavigate,
    useLocation: jest.fn()
}));
const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');

beforeEach(() => {
    useLocationMock.mockReturnValue({ ...addEmployeeMockData.LocationMockData, state: { action: 'add', bulkImportType: 'employees' } });
});

afterEach(() => {
    mock.reset();
    mockInsta.reset();
    useLocationMock.mockClear();
    cleanup();
});

describe('Bulk Import component', () => {
    it('should take snapshot', () => {
        const { container } = RTKRender(<BulkImport />);
        expect(container).toMatchSnapshot();
    });

    it('should component render properly', () => {
        RTKRender(<BulkImport />);
        expect(screen.getByText('Bulk Import')).toBeInTheDocument();
        expect(screen.getByText('Steps To Bulk Import Employees')).toBeInTheDocument();
        expect(
            screen.getByText('Make sure at least 1 employee with a Manager role is manually added to the application.')
        ).toBeInTheDocument();
        expect(screen.getByText('Start adding employees to the CSV template:')).toBeInTheDocument();
        expect(screen.getByText('Upload the CSV template.')).toBeInTheDocument();
        expect(screen.getByText('Click on the Start Import button.')).toBeInTheDocument();
    });

    it('should download template api run properly', async () => {
        mock.onGet('https://dummy-api.example.com/api/employees/template').replyOnce(200, '');
        RTKRender(<BulkImport />);
        fireEvent.click(screen.getByTestId('templateDownload'));
        await waitFor(() => expect(mock.history.get.length).toBe(1), { timeout: 3000 });
    });

    it('should download template with response work properly', async () => {
        mock.onGet('https://dummy-api.example.com/api/employees/template').replyOnce(
            200,
            `Employee Id, First Name, Last Name, Email Id, Contact No.(with Country Code), Gender(M/F/O), Date of Birth (DD-MM-YYYY), Active(Y/N), Department, Team, Designation, Role, Manager 1 Employee Id, Manager 2 Employee Id (Optional), Date of Joining (DD-MM-YYYY), Years of Experience (Years | Months), Consultant(Y/N)
        `
        );
        RTKRender(<BulkImport />);
        fireEvent.click(screen.getByTestId('templateDownload'));
        await waitFor(() => expect(mock.history.get.length).toBe(1), { timeout: 3000 });
    });

    it('should download template API error is handled properly', async () => {
        mock.onGet('https://dummy-api.example.com/api/employees/template').replyOnce(400, {});
        RTKRender(<BulkImport />);
        fireEvent.click(screen.getByTestId('templateDownload'));
        expect(await screen.findByText('Something went wrong.')).toBeInTheDocument();
    });

    it('should first time employee work properly', async () => {
        useLocationMock.mockReturnValue({
            ...addEmployeeMockData.LocationMockData,
            pathname: `${routeConstants.firstEmployeeBulkImportRedirect}`,
            state: { action: 'add', bulkImportType: 'employees' }
        });
        mock.onGet('https://dummy-api.example.com/api/employees/template').replyOnce(200, '');
        RTKRender(<BulkImport />);
        const backBtn = screen.getByText(/Go Back/i);
        await waitFor(() => expect(backBtn).toBeInTheDocument(), { timeout: 3000 });
        fireEvent.click(backBtn);
        await waitFor(() => expect(mockNavigate).toHaveBeenCalledWith(`${routeConstants.firstEmployee}`), { timeout: 3000 });
    });

    it('should kpi type render properly', async () => {
        useLocationMock.mockReturnValue({
            ...addEmployeeMockData.LocationMockData,
            state: { action: 'add', bulkImportType: 'kpis' }
        });
        mock.onGet('https://dummy-api.example.com/api/employees/template').replyOnce(200, '');
        RTKRender(<BulkImport />);
        expect(screen.getByText('Bulk Import')).toBeInTheDocument();
    });

    it('should upload and clear file work propely', async () => {
        RTKRender(<BulkImport />);
        const dropArea = screen.getByTestId('dropZone');
        const mockFile = new File(['employee name, id, contact'], 'test.csv', { type: 'text/csv' });
        fireEvent.change(dropArea, { target: { files: [mockFile] } });
        const importBtn = screen.getByTestId('importBtn');
        await screen.findByTestId('resetFile');
        expect(importBtn).toBeEnabled();
        fireEvent.click(screen.getByTestId('resetFile'));
        await waitFor(() => expect(importBtn).toBeDisabled());
        expect(screen.getByText('Drop File or Upload File here.')).toBeInTheDocument();
    });

    it('should show error when upload file has incorrect data', async () => {
        const mockFile = new File([''], 'test.csv', { type: 'text/csv' });
        mock.onPost('https://dummy-api.example.com/api/employees/upload').reply(400, { file: mockFile, message: '' });
        RTKRender(<BulkImport />);
        const dropArea = screen.getByTestId('dropZone');
        fireEvent.change(dropArea, { target: { files: [mockFile] } });
        const importBtn = screen.getByTestId('importBtn');
        await screen.findByTestId('resetFile');
        expect(importBtn).toBeEnabled();
        fireEvent.click(importBtn);
        expect(await screen.findByText('Unable to upload file due to error.')).toBeInTheDocument();
    });

    it('should upload file successfuly when data is correct', async () => {
        const csv = `Employee Id,First Name,Last Name,Email Id,Contact No.(with Country Code),Gender(M/F/O),Date Of Birth (DD-MM-YYYY),Active(Y/N),Department,Team,Designation,Role,Manager 1 Employee Id,Manager 2 Employee Id,Date Of Joining (DD-MM-YYYY),Year of Experience( Year | Months), Consultant(Y/N),
        SR003,P,S,ps@scalereal.com,+919876543210,M,12-10-1996,Y,Engineering,Team1,Des1,SDE,SR001,SR002,01-09-2022,2 | 11,N,`;
        const mockFile = new File([csv], 'test.csv', { type: 'text/csv' });
        mockInsta.onPost().reply(200, { file: mockFile, message: '' });
        RTKRender(<BulkImport />);
        const dropArea = screen.getByTestId('dropZone');
        fireEvent.change(dropArea, { target: { files: [mockFile] } });
        const importBtn = screen.getByTestId('importBtn');
        await screen.findByTestId('resetFile');
        expect(importBtn).toBeEnabled();
        fireEvent.click(importBtn);
        expect(await screen.findByText('Upload successful!')).toBeInTheDocument();
    });
});
