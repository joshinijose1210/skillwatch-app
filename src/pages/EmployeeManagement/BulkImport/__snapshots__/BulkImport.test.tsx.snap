// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bulk Import component should take snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Bulk Import
        </h3>
      </div>
    </div>
    <div
      class="sc-1q0dwkj-0 qUUwq"
    >
      <div
        class="sc-1q0dwkj-1 exGPty"
      >
        <div
          class="sc-1q0dwkj-2 beftPL"
        >
          <div
            class="sc-1q0dwkj-3 ebuXfU"
          >
            <h4
              class="sc-1ntgl3l-0 jPeJxZ"
              textalign="initial"
              textcolor="inherit"
              textvariant="h4"
              textweight="Medium"
            >
              Steps To Bulk Import Employees
            </h4>
            <div
              class="sc-1q0dwkj-4 dlzysz"
            >
              <svg
                class="sc-1pf3x8a-0 klDMsE"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M21.505 5.994l-9.88 9.88-3.48-3.47a.996.996 0 0 0-1.41 0 .996.996 0 0 0 0 1.41l4.18 4.18c.39.39 1.02.39 1.41 0l10.59-10.58a.996.996 0 0 0 0-1.41h-.01a.975.975 0 0 0-1.4-.01zm-4.24.01a.996.996 0 0 0-1.41 0l-5.64 5.64 1.41 1.41 5.64-5.65c.38-.38.38-1.02 0-1.4zm-12 12l-4.18-4.18c-.39-.39-.39-1.03 0-1.42a.996.996 0 0 1 1.41 0l4.88 4.9-.7.7a.996.996 0 0 1-1.41 0z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 dhgrkE"
                textalign="initial"
                textcolor="inherit"
                textvariant="body1"
                textweight="Regular"
              >
                Make sure at least 1 employee with a Manager role is manually added to the application.
              </span>
            </div>
            <div
              class="sc-1q0dwkj-4 dlzysz"
            >
              <svg
                class="sc-1pf3x8a-0 klDMsE"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M15 9.5h1.59c.89 0 1.33 1.08.7 1.71L12.7 15.8a.996.996 0 0 1-1.41 0L6.7 11.21c-.63-.63-.18-1.71.71-1.71H9v-5c0-.55.45-1 1-1h4c.55 0 1 .45 1 1v5zm-9 11c-.55 0-1-.45-1-1s.45-1 1-1h12c.55 0 1 .45 1 1s-.45 1-1 1H6z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 dhgrkE"
                textalign="initial"
                textcolor="inherit"
                textvariant="body1"
                textweight="Regular"
              >
                Download the CSV template.
              </span>
            </div>
            <div
              class="sc-1q0dwkj-4 dlzysz"
            >
              <svg
                class="sc-1pf3x8a-0 klDMsE"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10.99 8c0 1.66-1.33 3-2.99 3-1.66 0-3-1.34-3-3s1.34-3 3-3 2.99 1.34 2.99 3zm8 0c0 1.66-1.33 3-2.99 3-1.66 0-3-1.34-3-3s1.34-3 3-3 2.99 1.34 2.99 3zM8 13c-2.33 0-7 1.17-7 3.5V18c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-1.5c0-2.33-4.67-3.5-7-3.5zm7.03.05c.35-.03.68-.05.97-.05 2.33 0 7 1.17 7 3.5V18c0 .55-.45 1-1 1h-5.18c.11-.31.18-.65.18-1v-1.5c0-1.47-.79-2.58-1.93-3.41a.12.12 0 0 1-.01-.011.092.092 0 0 0-.03-.029z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 dhgrkE"
                textalign="initial"
                textcolor="inherit"
                textvariant="body1"
                textweight="Regular"
              >
                Start adding employees to the CSV template:
                <ul
                  class="sc-1q0dwkj-15 lbKKyb"
                >
                  <li
                    class="sc-1q0dwkj-16 edaEkd"
                  >
                    While adding the employees, make sure employees with Manager roles are added first. 
                  </li>
                  <li
                    class="sc-1q0dwkj-16 edaEkd"
                  >
                    Enter valid data for all the employees 
                  </li>
                  <li
                    class="sc-1q0dwkj-16 edaEkd"
                  >
                    You can set the status of the employee as Active by entering Y in the ‘Active’ column. 
                  </li>
                  <li
                    class="sc-1q0dwkj-16 edaEkd"
                  >
                    You can set the status of the employee as Inactive by entering N in the ‘Active’ column. 
                  </li>
                  <li
                    class="sc-1q0dwkj-16 edaEkd"
                  >
                    You can set the role according to the roles listed in the application (example: HR, Manager, Employee), in the ‘Role’ column depending on the different module/feature permissions you would like to give an employee.
                  </li>
                  <li
                    class="sc-1q0dwkj-16 edaEkd"
                  >
                    Maximum 500 employees can be added at a time. 
                  </li>
                  <li
                    class="sc-1q0dwkj-16 edaEkd"
                  >
                    Once you enter all the valid data, please save the CSV template. 
                  </li>
                </ul>
              </span>
            </div>
            <div
              class="sc-1q0dwkj-4 dlzysz"
            >
              <svg
                class="sc-1pf3x8a-0 klDMsE"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14 16.296h-4c-.55 0-1-.45-1-1v-5H7.41c-.89 0-1.33-1.08-.7-1.71l4.59-4.59a.996.996 0 0 1 1.41 0l4.59 4.59c.63.63.18 1.71-.71 1.71H15v5c0 .55-.45 1-1 1zm4 2H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 dhgrkE"
                textalign="initial"
                textcolor="inherit"
                textvariant="body1"
                textweight="Regular"
              >
                Upload the CSV template.
              </span>
            </div>
            <div
              class="sc-1q0dwkj-4 dlzysz"
            >
              <svg
                class="sc-1pf3x8a-0 klDMsE"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M8.551 6v3.74c-1.21-.81-2-2.18-2-3.74 0-2.49 2.01-4.5 4.5-4.5s4.5 2.01 4.5 4.5c0 1.56-.79 2.93-2 3.74V6a2.5 2.5 0 0 0-5 0zm4.61 6c.31 0 .61.07.89.21l4.08 2.03c.77.38 1.21 1.22 1.09 2.07l-.63 4.47c-.14.99-.98 1.72-1.98 1.72h-6.1c-.53 0-1.04-.21-1.42-.58l-4.01-4.01c-.43-.44-.43-1.14 0-1.58.27-.27.66-.39 1.03-.31l3.44.72V6c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5v6h.61z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 dhgrkE"
                textalign="initial"
                textcolor="inherit"
                textvariant="body1"
                textweight="Regular"
              >
                Click on the Start Import button.
              </span>
            </div>
            <div
              class="sc-1q0dwkj-4 dlzysz"
            >
              <svg
                class="sc-1pf3x8a-0 klDMsE"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 13c0 .55-.45 1-1 1H5c-.55 0-1-.45-1-1V8l6.94 4.34c.65.41 1.47.41 2.12 0L20 8v9zM4 6l8 5 8-5H4z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 dhgrkE"
                textalign="initial"
                textcolor="inherit"
                textvariant="body1"
                textweight="Regular"
              >
                Auto generated Invitation Email will be sent to employees.
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-1q0dwkj-6 cssxYE"
        >
          <div
            class="sc-1q0dwkj-4 dlzysz"
          >
            <button
              class="sc-1k2druj-0 dsEKPA"
              data-testid="templateDownload"
              edges="square"
              type="button"
              variant="outlined"
            >
              <svg
                class="sc-1pf3x8a-0 klDMsE"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M15 9.5h1.59c.89 0 1.33 1.08.7 1.71L12.7 15.8a.996.996 0 0 1-1.41 0L6.7 11.21c-.63-.63-.18-1.71.71-1.71H9v-5c0-.55.45-1 1-1h4c.55 0 1 .45 1 1v5zm-9 11c-.55 0-1-.45-1-1s.45-1 1-1h12c.55 0 1 .45 1 1s-.45 1-1 1H6z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-1ntgl3l-0 eOsHHD"
                textalign="initial"
                textcolor="inherit"
                textvariant="button1"
              >
                Download csv template
              </span>
            </button>
          </div>
          <div
            class="sc-1q0dwkj-7 gHQAIr"
          >
            <div
              class="sc-1q0dwkj-8 bSYzhS"
              role="presentation"
              tabindex="0"
            >
              <img
                alt="upload"
                class="sc-1q0dwkj-12 hKliqI"
                src="[object Object]"
              />
              <label
                class="sc-1q0dwkj-10 bTRaqw"
                id="button"
              >
                 Drop File or Upload File here.
              </label>
              <input
                accept="text/csv,.csv"
                class="sc-1q0dwkj-9 hdfMmW"
                data-testid="dropZone"
                multiple=""
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
                tabindex="-1"
                type="file"
              />
            </div>
            
            <button
              class="sc-1k2druj-0 eLgfbc"
              data-testid="importBtn"
              disabled=""
              edges="square"
              type="button"
              variant="solid"
            >
              <span
                class="sc-1ntgl3l-0 eOsHHD"
                textalign="initial"
                textcolor="inherit"
                textvariant="button1"
              >
                Start Import
              </span>
            </button>
            <em>
              (Only .csv file will be accepted)
            </em>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
